corehq/apps/cloudcare/static/cloudcare/js/form_entry/webformsession.js

```diff
@@ -444,20 +448,23 @@ hqDefine("cloudcare/js/form_entry/webformsession", function () {
             form.isSubmitting(true);
             var submitAttempts = 0,
                 timer = setInterval(function () {
-                    var answers;
                     if (form.blockSubmit() && submitAttempts < 10) {
                         submitAttempts++;
                         return;
                     }
                     clearInterval(timer);
 
-                    answers = accumulateAnswers(form);
-                    self.serverRequest(
-                        {
+                    var requestCallback = function () {
+                        var answers = accumulateAnswers(form);
+                        return {
                             'action': Const.SUBMIT,
                             'answers': answers,
                             'prevalidated': prevalidated,
-                        },
+                        }
+                    }
+                    requestCallback.action = Const.SUBMIT;
```

Comment:
I think adding a attribute to a function is javascript pattern okay?

Reply:
Yeah, I think this is odd but okay. The only alternative I can think of is passing `action` as a separate parameter instead of as part of `requestParams`, then adding it to `requestParams` inside of `serverRequest`/`_serverRequest`.

Reply:
This is actually what I did originally before switching to this, but it involved changing all of the things that called `serverRequest` and it felt very clunky.